xds/src/main/java/io/grpc/xds/OrcaMetricReportingServerInterceptor.java

```diff
@@ -79,9 +79,14 @@ public void close(Status status, Metadata trailers) {
             super.close(status, trailers);
           }
         };
-    final CallMetricRecorder recorder = InternalCallMetricRecorder.newCallMetricRecorder();
+    Context ctx = Context.current();
+    if (InternalCallMetricRecorder.CONTEXT_KEY.get(ctx) == null) {
+      ctx =
+          ctx.withValue(InternalCallMetricRecorder.CONTEXT_KEY,
+              InternalCallMetricRecorder.newCallMetricRecorder());
+    }
```

Comment:
It's better to save the instance here and pass it to the forwarding call, instead of forwarding call grabbing it from current Context. This is because application has the freedom to **not** call `ServerCall.close()` from the original Context.

Reply:
Done.